### PR TITLE
Add app note regarding using tams for data

### PR DESCRIPTION
**Pull request by j616**
# Details
Add app note regarding using tams for data

# Pivotal Story (if relevant)
Story URL: https://www.pivotaltracker.com/story/show/187412251

# Related PRs
_Where appropriate. Indicate order to be merged._

# Submitter PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] API version has been incremented if necessary
- [ ] ADR status has been updated, and ADR implementation has been recorded
- [ ] Documentation updated (README, etc.)
- [ ] PR added to Pivotal story (if relevant)
- [ ] Follow-up stories added to Pivotal

# Reviewer PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] Design makes sense, and fits with our current code base
- [ ] Code is easy to follow
- [ ] PR size is sensible
- [ ] Commit history is sensible and tidy

# Info on PRs
The checks above are guidelines. They don't all have to be ticked, but they should all have been considered.

### CODE REVIEW

File: docs/appnotes/0004-tams-for-data.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+But it might not be an efficient storage mechanism.
+Typically, the content being stored should be larger than the metadata stored alongside it.
+
+## Examples
+
+### Timed Text
+
+Timed text, such as closed captions, is relatively low data rate.
+But it is often used alongside audio and video, using similar workflows.
+Existing live workflows in television broadcast facilities may make use of [SMPTE 2110-43](https://ieeexplore.ieee.org/document/9521125)/[RFC8759](https://www.rfc-editor.org/rfc/rfc8759.html) to carry small independently XML documents containing the captions.
```
Comment (samdbmg) [inline docs/appnotes/0004-tams-for-data.md:32]: Nit: Is there a word missing after "independently"?

File: docs/appnotes/0004-tams-for-data.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+The usage patterns, recommended frequency, and likely size of these documents lend themselves well to storage in a TAMS store.
```
Comment (samdbmg) [inline docs/appnotes/0004-tams-for-data.md:33]: It might be worth spelling this out a bit more completely
```suggestion
The usage patterns, recommended frequency, and likely size of these documents lend themselves well to storage in a TAMS store, however care should be taken to balance the flexibility of shorter segments (and smaller XML documents) against the increased metadata storage overhead.
```

File: docs/appnotes/0004-tams-for-data.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+### Time-based metadata
+
+Consider a workflow that records who appears on screen at particular times.
+This type of data likely wouldn't be appropriate to store in TAMS, as you would not be able to efficiently perform basic queries such as what times an individual appeared on screen.
+
+Time Addressable Metadata is an important complimentary concept to Time Addressable Media.
+But it is not within the scope of TAMS.
```
Comment (samdbmg) [inline docs/appnotes/0004-tams-for-data.md:41]: Nit: Specificity about what it's in scope for. 
```suggestion
But it is not within the scope of this repository or specification.
```

It's of interest to us in a bigger picture sense though, so you could add a "we'd be happy to chat" throwaway comment though.

File: docs/appnotes/0004-tams-for-data.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+Consider a system which records loudness levels of audio a small number of times a second.
+If a segment length of ~1 second was used, common for media stored in TAMS to enable efficient random access, then the segment metadata may be many times larger than the segment content.
+The underlying file object store may end up storing a large number of very small objects.
+This is often inefficient both in resources, and cost.
+Using larger segments would bring about different issues.
+Segments normally only become available once they are complete.
+A large segment size will, therefore, increase the latency in segments becoming available.
+TAMS would not be appropriate for such low rate data.
+
+### High data rate production data
```
Comment (samdbmg) [inline docs/appnotes/0004-tams-for-data.md:54]: This is a really good example 💯

File: docs/appnotes/0004-tams-for-data.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Considerations
+
+### Indexing of content
+
+TAMS indexes content based on a small number of specific types of metadata.
+Content may be queried based on these types of metadata (e.g. Flow/Source ID, tags, time-ranges etc).
+These metadata apply to the Source/Flow as a whole.
+It is not possible to query the data itself, or metadata as it applies to a portion of the Source/Flow.
+As such, TAMS is usually not appropriate for the storing of timeline-based metadata.
```
Comment (samdbmg) [inline docs/appnotes/0004-tams-for-data.md:18]: I can see what this is trying to say, but found it hard to parse until I'd read the whole document. How about this?

```suggestion
TAMS indexes Flows as a whole using a small number of pieces of metadata (Flow/Source ID, tags, etc.), however the only index that can be used inside a specific Flow is time.
It is not possible to query the data/content or metadata as it applies to a portion of the Source/Flow, which would be needed to query timeline-based metadata (e.g. it's not possible to query a transcript for occurrences of a specific word using TAMS).
As such, TAMS is usually not appropriate for the storing of timeline-based metadata.
However some examples of where it may be appropriate are given below.
```

**Review by samdbmg — APPROVED** [review]
LGTM overall - a few nits inlined

File: docs/appnotes/0004-tams-for-data.md (hunk 1 above)
Comment (j616) [inline docs/appnotes/0004-tams-for-data.md:32]:
```suggestion
Existing live workflows in television broadcast facilities may make use of [SMPTE 2110-43](https://ieeexplore.ieee.org/document/9521125)/[RFC8759](https://www.rfc-editor.org/rfc/rfc8759.html) to carry small independently decodable  XML documents containing the captions.
```
